core/models.py

```diff
@@ -1058,12 +1058,25 @@ def clean(self):
             raise ValidationError(_("You must provide a file"))
 
     def apply_rights_recursively(self, *, only_folders=False):
-        children = self.children.all()
-        if only_folders:
-            children = children.filter(is_folder=True)
-        for c in children:
-            c.copy_rights()
-            c.apply_rights_recursively(only_folders=only_folders)
+        file_ids = []
+        explored_ids = [self.id]
+        while len(explored_ids) > 0:  # find all children recursively
+            file_ids.extend(explored_ids)
+            next_level = SithFile.objects.filter(parent_id__in=explored_ids)
+            if only_folders:
+                next_level = next_level.filter(is_folder=True)
+            explored_ids = list(next_level.values_list("id", flat=True))
+        for through in (SithFile.view_groups.through, SithFile.edit_groups.through):
+            groups = list(
```

Comment:
Y'a vraiment besoin de forcer la requête ici avec `list()`? C'est du pinaillage, mais pourquoi pas juste laisser l'itérateur se consommer juste après?

Reply:
Je sais pas trop pourquoi, mais quand on ne force fetch pas, alors la variable `groups` ne yield rien. Ce qui fait qu'on ne crée rien et que ça ne marche pas.